staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_external.go

```diff
@@ -591,16 +591,28 @@ func (g *Cloud) updateTargetPool(loadBalancerName string, hosts []*gceInstance)
 		toRemove = append(toRemove, &compute.InstanceReference{Instance: link})
 	}
 
-	if len(toAdd) > 0 {
-		if err := g.AddInstancesToTargetPool(loadBalancerName, g.region, toAdd); err != nil {
+	for len(toAdd) > 0 {
+		// Do not remove more than maxInstancesPerTargetPoolUpdate in a single call.
+		instancesCount := len(toAdd)
+		if instancesCount > maxInstancesPerTargetPoolUpdate {
+			instancesCount = maxInstancesPerTargetPoolUpdate
+		}
+		if err := g.AddInstancesToTargetPool(loadBalancerName, g.region, toAdd[:instancesCount]); err != nil {
```

Comment:
I guess this operation blocks long enough to avoid QPS issues?

Reply:
Yes - it's pretty long operation (for 1k instance can take up to minutes) - added a comment.